test/test_runner/base_test_runner.go

```diff
@@ -33,6 +33,7 @@ type ITestRunner interface {
 	GetMeasuredMetrics() []string
 	SetupBeforeAgentRun() error
 	SetupAfterAgentRun() error
+	RunAgent(runner *TestRunner) (status.TestGroupResult, error)
```

Comment:
I do really think it's weird to make the TestRunner interface take an implementation of itself as an argument for a method. I'll leave this be for now, but this is due for an overhaul

Reply:
Lint failures fixed. I agree this is a weird approach but I think we would need to change the way we structure test_runners in order for this to be more straightforward.